.github/build-iroha1.src.yml

```diff
@@ -0,0 +1,540 @@
+## TODO 3. More powerful Mac machine to reduce build time from 40min to 10min with hot vcpkg cache, and from 2hrs to 27min without cache
+##         GitHub's default runners also idles much time before started
+## TODO 4. Cancel previous runs if their duration is less than 10 minutes, protect almost done builds from being killed
+## TODO 6. Do not rebuild vcpkg tool every time [takes about 1min], see vcpkg/bootstrap
+## FIXME 7. ctest integration_fake_peer_example_test
+## FIXME 8. cannot push docker image iroha-builder without credentials when PR is from fork.
+##       Because GitHub secrets inaccessable to PRs from forks.
+##       Solution: open PR from the same repo.
+## FIXME 9. change :container: to hyperledger/iroha-builder after first one pushed instead of ikyb/iroha-builder
+## TODO optimization cache between builds -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+## TODO store ccache with rclone
+## TODO Use vcpkg binarycaching on CI https://devblogs.microsoft.com/cppblog/vcpkg-accelerate-your-team-development-environment-with-binary-caching-and-manifests/
+
+## TODO schedule: nightly
+
+#${{ github.event.pull_request.head.repo.fork }}   ## check if PR head repo is fork
+## ternary ${{ fromJSON('["no", "yes"]')[github.ref != 'refs/heads/master'] }}
+
+
+on:
+  push:
+    branches: [ main, support/1.2.x ]
+    tags: [ 'v*' ]
+  pull_request:
+    branches: [ main, support/1.2.x ]  ## target branches
+  workflow_dispatch:
+    ## NOTE: Able to run via cmdline: gh workflow run Iroha1
+    inputs:
+      whatever:
+        description: 'some-string.'
+        required: true
+        default: 'hello world!'
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  pr_comment_reaction_rocket:
+    ## Just to react to valid comment with rocket
+    runs-on: ubuntu-latest
+    if: ${{ github.event.comment &&
+          github.event.issue.pull_request &&
+          startsWith(github.event.comment.body, '/buildme') }}
+    steps:
+      - &step_show_context
+        name: Show context
+        run: |
+          cat >/dev/null <<'END'
+            ${{ toJson(github) }}
+          END
+      -
+        name: Reaction
+        run: |
+          # send reaction to comment to show build was triggered
+          curl ${{github.event.comment.url}}/reactions \
+            -X POST \
+            -d '{"content":"rocket"}' \
+            -H "Accept: application/vnd.github.squirrel-girl-preview+json" \
+            -H "Authorization: token ${{github.token}}"
+
+
+  Docker-iroha-builder:
+    if: ${{false}}
+    # ## Check if PR same repo, so GitHub secrets are available, without them nothing to do
+    # if: ${{ github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name }}
+    runs-on: [ self-hosted, Linux ]
+    continue-on-error: true  ##FIXME  ##Sets needs.job_id.result always to success
+    steps:
+      - *step_show_context
+      - &step_cancel_previous_runs
+        if: ${{false}}
```

Comment:
Add Comment or TODO why this is disabled

Reply:
done